### PR TITLE
[Dynamic Spec Decoding] Minor fix for disabling speculative decoding

### PR DESCRIPTION
**Pull request by LiuXiaoxuanPKU**
In the current implementation, even if `running_queue_size >= speculative_disable_by_batch_size`, it will still go through the speculative decoding logic, which includes get_spec_proposals with k=0, score_proposal, rejection sampling and create sampler output. The flow introduces extra overhead (especially rejection sampling), which makes disabling speculative decoding slower than 'real' without speculative decoding. 

To fix this, we can just reuse the `_run_no_spec` to avoid touching the sd flow at all. Also add a test to check the correctness.

Concretely, for a batch size of 8 with 128 output tokens, TP=4, for LLama3-70B, the batch latency is
| Without SD | Disable SD | Disable SD after fix |
| -- | -- | -- |
| 5.4 s            |  5.8s            | 5.5s                          |    

Here, without SD means not using sd flag at all. Disable SD means using SD flag, but set `speculative_disable_by_batch_size` smaller than batch size to disable speculative decoding.
After the fix, we are still slower than the original case, this is caused by broadcasting control flow, which will be fixed in future PRs.

### CODE REVIEW

File: vllm/spec_decode/spec_decode_worker.py [hunk 1]
```diff
@@ -276,7 +276,8 @@ def execute_model(
         # If no spec tokens, call the proposer and scorer workers normally.
         # Used for prefill.
```
Comment (comaniac) [inline vllm/spec_decode/spec_decode_worker.py:277]: Refine the comment to include auto disable?

**Review by comaniac — APPROVED** [review]
Thanks for the fix!

File: tests/spec_decode/e2e/test_ngram_correctness.py [hunk 2]
```diff
@@ -170,3 +170,44 @@ def test_ngram_different_k(baseline_llm_generator, test_llm_generator,
...
+@pytest.mark.parametrize("per_test_common_llm_kwargs", [{}])
+@pytest.mark.parametrize("baseline_llm_kwargs", [{}])
+@pytest.mark.parametrize("test_llm_kwargs",
+                         [{
+                             "speculative_model": "[ngram]",
+                             "num_speculative_tokens": 5,
+                             "ngram_prompt_lookup_max": 3,
+                             "speculative_disable_by_batch_size": 4
+                         }])
+@pytest.mark.parametrize("batch_size", [1, 2, 5, 8])
```
Comment (cadedaniel) [inline tests/spec_decode/e2e/test_ngram_correctness.py:195]: nit: suggest only [1, 5] to reduce test time

**Review by cadedaniel — APPROVED** [review]
LGTM, let's update the comment before merging

**Comment by comaniac** [comment]
Looks like the CI failure is unrelated and we should just merge this. cc @simon-mo

**Pull request merged**